### PR TITLE
Dbeaver/dbeaver#35882 fix sql script execution

### LINKED ISSUE #35882: Execution of the script on disconnected state
### Description

1. Create a script with the following query (PostgreSQL)
```
create or replace procedure proc1()
language plpgsql
as
$$
begin
	select pg_sleep(10000);
end;
$$
;
```
2. Disconnect from the database
3. Execute this query and get an error - we didn't correctly get the query end
4. Execute this query again while you're connected to the database - you don't get this error
image

### DBeaver Version

DBeaver 24.2.2

### Operating System

_No response_

### Database and driver

_No response_

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HocKu7**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLParserContext.java [hunk 1]
```diff
@@ -74,7 +103,24 @@ public IDocument getDocument() {
     }
 
     public SQLDialect getDialect() {
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/SQLParserContext.java:105]: By the name, it looks like a getter, but it instantiates an uninitialized script dialect. Getters should not do such things.

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorBase.java [hunk 2]
```diff
@@ -1309,6 +1323,12 @@ protected boolean isNavigationTarget(Annotation annotation) {
         return super.isNavigationTarget(annotation);
     }
 
+    @Nullable
+    @Override
+    public DBPDataSourceContainer getDataSourceContainer() {
+        return null;
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorBase.java]: Maybe I missed something, but I don't understand the logic here. You call this method from other methods of this class, however, it always returns null. What's the point of doing that?

Comment (HocKu7) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorBase.java]: It's overridden in SQLEditor

**Comment by MashaKorax** [comment]
verified

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorBase.java (hunk 2 above)
Comment (serge-rider) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/SQLEditorBase.java]: Logic is still incorrect. Why SQLEditorBase become DBPDataSourceContainerProvider ?
It mustn't return null, at least it should return dataSource.

**Review by serge-rider — CHANGES_REQUESTED** [review]
https://github.com/dbeaver/dbeaver/pull/36056#discussion_r1818833826

**Comment by kseniaguzeeva** [comment]
Verified

**Pull request merged**